### ISSUE TITLE
App deactivates after some time

### ISSUE THREAD

**Issue by CrimsonFork** [SimpleMobileTools/Simple-Clock#211]
This is a stable issue on my device. Approx. 24 hours after using it (interacting with the alarm interface doesn't count), it becomes inactive, but restores activity once it's opened again.

I have tried disabling battery optimizations, but no avail.

My device is a Moto E⁴ Plus (XT1771) with its stock rom (A 7.1.1) and some things manually removed (can't tell for sure anymore, but mostly just vendor and Google bloatware).

**Comment by tibbi** [comment]
what app version are you using? Do other apps work well?

**Comment by MrHeating** [comment]
I'm on F-Droid's latest 5.2.2 build. All apps' direct functionality is ok, and I believe the Calender's notifications work just fine, although, I barely use them, so can't tell for sure.

On 26 June 2020 10:37:53 GMT+02:00, Tibor Kaputa <notifications@github.com> wrote:
>what app version are you using? Do other apps work well?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/SimpleMobileTools/Simple-Clock/issues/211#issuecomment-650060239
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by tibbi** [comment]
so your issue is that alarms wont sound after some time? Or the whole app just freezes?

**Comment by MrHeating** [comment]
I'm not sure if "freezing" is the right word, more like ending its process out of the blue. The alarm icon in the notivication bar also vanishes. When opening again everything returns to normal.

Hence why battery optimizations wrre my first thought.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by webartifex** [comment]
Hello, I feel I also keep seeing this bug.

I installed Simple Clock on three devices in the past 2 weeks (all latest version from F-Droid store): Samsung Galaxy S21, Samsung Galaxy Tab S7 Plus, and Moto G7 Power.

I added a clock widget on the home screen.

In the beginning, everything works as expected.

After some time, the times shown in the widget somehow "freeze" (so, it is clearly showing some time from the past).

After opening the clock app, the widget updates and shows the correct and current time. So, it seems like the clock widget simple "deactivates" or "freezes".
Interestingly, that also happens sometimes for the torch widget (one click button on home screen to enable/disable the flashlight).

On both Samsung devices, I made sure that Simple Clock is on the "Never sleep" list of apps that are not distracted in any form, even with power saving mode.

I am not an Android dev but if you tell me how I can provide debug info, I am sure I can get this to work (I am a Python backend dev) and post you further info.

Let me know how else I can help. Great effort with all the simplemobiletools apps.
